client/blocks/site-preview/index.jsx

```diff
@@ -1,110 +1,82 @@
-import debugFactory from 'debug';
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import WebPreview from 'calypso/components/web-preview';
 import { addQueryArgs } from 'calypso/lib/route';
 import isDomainOnlySite from 'calypso/state/selectors/is-domain-only-site';
 import { getSiteOption, getSiteSlug, isSitePreviewable } from 'calypso/state/sites/selectors';
 import { getCurrentLayoutFocus } from 'calypso/state/ui/layout-focus/selectors';
 import { closePreview } from 'calypso/state/ui/preview/actions';
-import {
-	getPreviewSite,
-	getPreviewSiteId,
-	getPreviewUrl,
-} from 'calypso/state/ui/preview/selectors';
+import { getPreviewSiteId, getPreviewUrl } from 'calypso/state/ui/preview/selectors';
 
-const debug = debugFactory( 'calypso:site-preview' );
+function usePreviewCounter( siteId, showPreview ) {
+	const [ previewCount, setPreviewCount ] = useState( 0 );
 
-class SitePreview extends Component {
-	static propTypes = {
-		className: PropTypes.string,
-		showPreview: PropTypes.bool,
-		previewUrl: PropTypes.string,
-		selectedSite: PropTypes.object,
-		selectedSiteId: PropTypes.number,
-		selectedSiteNonce: PropTypes.string,
-		selectedSiteUrl: PropTypes.string,
-		closePreview: PropTypes.func.isRequired,
-	};
+	useEffect( () => {
+		if ( siteId ) {
+			setPreviewCount( 0 );
+		}
+	}, [ siteId ] );
 
-	state = {
-		previewCount: 0,
-	};
+	useEffect( () => {
+		if ( showPreview ) {
+			setPreviewCount( ( n ) => n + 1 );
+		}
+	}, [ showPreview ] );
 
-	previewCounter = 0;
+	return previewCount;
+}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.selectedSiteId && this.props.selectedSiteId !== nextProps.selectedSiteId ) {
-			this.previewCounter = 0;
-		}
+function SitePreview( { className } ) {
+	const dispatch = useDispatch();
 
-		if ( ! this.props.showPreview && nextProps.showPreview ) {
-			debug( 'forcing refresh' );
-			this.previewCounter > 0 && this.setState( { previewCount: this.previewCounter } );
-			this.previewCounter += 1;
-		}
+	const selectedSiteId = useSelector( getPreviewSiteId );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, selectedSiteId ) );
+	const selectedSiteUrl = `https://${ siteSlug }`.replace( /::/g, '/' );
```

Comment:
`siteSlug.replace()` is more precise, we're replacing only in the slug, never in the `https://` previx.

Reply:
Anyway, I think `selectedSiteUrl` is never really used and can be removed. When showing the preview, we always dispatch the `setPreviewUrl` action, setting a valid `previewUrl`.